src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2314,23 +2314,22 @@ public static unsafe Vector128<uint> Narrow(Vector128<ulong> lower, Vector128<ul
         /// <param name="indices">The per-element indices used to select a value from <paramref name="vector" />.</param>
         /// <returns>A new vector containing the values from <paramref name="vector" /> selected by the given <paramref name="indices" />.</returns>
         [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector128<byte> Shuffle(Vector128<byte> vector, Vector128<byte> indices)
         {
-            Unsafe.SkipInit(out Vector128<byte> result);
-
-            for (int index = 0; index < Vector128<byte>.Count; index++)
+            if (ShuffleUnsafeIntrinsicSupported)
```

Comment:
This is a lot of logic for something that is meant to be a cold path and which ideally would be handled in the JIT instead.

Reply:
I'll have a look into that some time next week probably :) thanks